### PR TITLE
fix(vite): add `file://` protocol to `vite-node` entry file path

### PR DESCRIPTION
**Pull request by jremye**
### 🔗 Linked issue
#4965

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/vite/src/vite-node.ts [hunk 1]
```diff
@@ -87,6 +89,7 @@ export async function prepareDevServerEntry (ctx: ViteBuildContext) {
   const host = ctx.nuxt.options.server.host || 'localhost'
   const port = ctx.nuxt.options.server.port || '3000'
   const protocol = ctx.nuxt.options.server.https ? 'https' : 'http'
+  const winFileProtocol = isWindows ? 'file://' : ''
```
Comment (antfu) [inline packages/vite/src/vite-node.ts:92]: We could just always include the protocol

Comment (jremye) [reply packages/vite/src/vite-node.ts]: Thanks, I have edited

**Comment by danielroe** [comment]
Would it be worth using `pathToFileURL` instead?

File: packages/vite/src/vite-node.ts [hunk 2]
```diff
@@ -99,10 +99,10 @@ export async function prepareDevServerEntry (ctx: ViteBuildContext) {
 
   await fse.writeFile(
     resolve(ctx.nuxt.options.buildDir, 'dist/server/server.mjs'),
-    `export { default } from ${JSON.stringify(resolve(distDir, 'runtime/vite-node.mjs'))}`
+    `export { default } from ${JSON.stringify('file://' + resolve(distDir, 'runtime/vite-node.mjs'))}`
```
Comment (pi0) [inline packages/vite/src/vite-node.ts:102]: We might use [pathToFileURL](https://nodejs.org/api/url.html#urlpathtofileurlpath)

**Review by pi0 — APPROVED** [review]
Thanks!

**Pull request merged**